### ISSUE TITLE
overwriting font-family

### ISSUE THREAD

**Issue by noorzaie** [vuetifyjs/vuetify#8169]
There is many inline font-families and so it's not possible to overwrite them using sass variables.

![image](https://user-images.githubusercontent.com/7344137/62107184-4c95eb00-b2bc-11e9-8197-732c6b847b9c.png)

**Comment by ThomasKuhlmann** [comment]
@noorzaie Probably better to ask this in the discord channel. Anyway, here's how you change the font:

Add custom sass file as per the instructions in the docs:
https://vuetifyjs.com/en/customization/sass-variables

Then add this in your sass file

```
@import "~vuetify/src/styles/styles.sass";
@import url("https://fonts.googleapis.com/css?family=Maven+Pro:400,500,700,900&display=swap");
$body-font-family: "Maven Pro" !important;

```

And replace the font with whatever font you want.

**Comment by noorzaie** [comment]
This just changes global font family. But the problem I've faced is inline styles. You can see inline font-family in image I've attached. So overwriting `$body-font-family` won't work in this case.
The value of `$body-font-family` is `"Roboto", sans-serif !default` and is different from inline styles, So I think inline font-families are different from the sass variable.

**Comment by jacekkarczmarczyk** [comment]
https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/styles/elements/_typography.sass#L42

**Comment by filipyoo** [comment]
Reopened this because I got the same issue, the font doesn't apply to headings (h1, h2...) tags... So when using typography with class like `title` (which is `h6`) the custom font doesn't get applied...
A comment in SO also stated it:
https://stackoverflow.com/questions/45598884/change-default-font-in-vuetify

**Comment by andrewsali** [comment]
I had to declare my custom font-family **before** importing the default vuetify style in my custom main.scss. That way vuetify properly picks up the custom definition:

```
$body-font-family: ...;

@import "~vuetify/src/styles/styles.sass";
```

**Comment by filipyoo** [comment]
Thank you for your reply!
Unfortunately it still doesn't work for me...
I did the way you described, and in my `nuxt.config.js` I have:
```
  head: {
    ...
    link: [
      {
        rel: 'stylesheet',
        href: 'https://fonts.googleapis.com/css?family=Varela+Round'
      }
    ]
  },
  /*
   ** Global CSS
   */
  css: ['~/assets/scss/main.scss'],
  ...
  modules: [
    '@nuxtjs/style-resources',
  ],
  styleResources: {
    scss: ['~/assets/scss/variables.scss']
  },
  ...
  vuetify: {
    customVariables: ['~/assets/scss/variables.scss'],
    theme: {
    ...
      }
    }
  },
```

And in my `~/assets/scss/main.scss` file:
```
$body-font-family: 'Varela Round';
$heading-font-family: 'Varela Round'; // With or without this line, it still doesn't work

@import '~vuetify/src/styles/styles.sass';
```

**Comment by pimboden** [comment]
None of the solutions work:
I have nuxt.config with 
 
     buildModules: [
        // https://github.com/nuxt-community/vuetify-module#readme
        '@nuxtjs/vuetify'
     ...
    ]
    ...
    vuetify: {
    customVariables: ['~/assets/scss/vuetifyvariables.scss'],
     theme: {
       dark: false
    }
    ....

and in my ~/assets/scss/vuetifyvariables.scss I wrote:
 
     $body-font-family: 'FSAlbertWeb-Regular',Helvetica,sans-serif;
     @import "~vuetify/src/styles/styles.sass";

Still I see:

    .v-application {
        font-family: "Roboto", sans-serif;
        line-height: 1.5;
    }

**Comment by leasontou** [comment]
Has anyone solved this problem?

**Comment by pimboden** [comment]
No.. They "solved" it by closing the issue without a solution. I had to overwrite it by css using important!

**Comment by leasontou** [comment]
> No.. They "solved" it by closing the issue without a solution. I had to overwrite it by css using important!

emmm... I think I will follow your steps.

**Comment by JosephSaw** [comment]
> I had to declare my custom font-family **before** importing the default vuetify style in my custom main.scss. That way vuetify properly picks up the custom definition:
> 
> ```
> $body-font-family: ...;
> 
> @import "~vuetify/src/styles/styles.sass";
> ```

In case anyone else is reading this, this definitely worked for me.

**Comment by pwnz22** [comment]
I also cant change the font family. I have this:

```
@import './fontface.scss';

$body-font-family: 'GT-Walsheim-Regular', sans-serif;
$heading-font-family: 'GT-Walsheim-Regular', sans-serif;

@import '~vuetify/src/styles/settings/_variables';

@each $heading, $style in $headings {
    $headings: map-merge($headings, ($heading: map-merge($style, ('font-family': $heading-font-family))));
}
```

Anyone can help?

**Comment by alvaro-canepa** [comment]
Is there any working example of how to replace main fonts?

**Comment by imperatormk** [comment]
Worked for me!

nuxt.config.js

`vuetify: {
    customVariables: ['~/assets/variables.scss'],
    theme: {
      dark: false
    }
  },`

variables.scss

`$body-font-family: Montserrat;

.v-application {
  .display-1,
  .display-2,
  .display-3,
  .display-4,
  .headline,
  .title,
  .subtitle-1,
  .subtitle-2,
  .body-1,
  .body-2,
  .caption,
  .overline {
    font-family: $body-font-family !important;
  }
}`

**Comment by loomchild** [comment]
@filipyoo, @pimboden: Perhaps you are missing `treeShake: true` in `nuxt.config.js` file `vuetify` section?

I wrote a short article explaining the solution, there's also a complete code example at the end:
https://medium.com/untitled-factory/changing-default-font-in-vuetify-js-and-nuxt-js-3894e726ff10

**Comment by indigoswann** [comment]
Re-open this, it's not fixed.  It's literally impossible to override the default Roboto font.

**Comment by joaokrabbe** [comment]
Same here, trying everthing. Nothing work.

**Comment by joaokrabbe** [comment]
I find a workaround!

I tried to use custom variables, but the import `import 'vuetify/dist/vuetify.min.css';` was overwriting my definitions.
So I appealed to css specificity, and set my overwriting with css id, and it's beats the class overwrite from vuetify.min.css.

**_If anyone needs:_**
`@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,400i,600,700,800&display=swap');

// Globals
$body-font-family: 'Open Sans';

$typoOptions: display-4, display-3, display-2, display-1, headline, title, subtitle-1, subtitle-2, body-1, body-2, caption, overline;
#app {
  font-family: $body-font-family, sans-serif !important;
  @each $typoOption in $typoOptions {
    .#{$typoOption} {
      font-family: $body-font-family, sans-serif !important;
    }
  }
}`

**ps.: I install vuetify using vue cli 3**

**Comment by BonBonSlick** [comment]
Salvation is easy peasy, do not use this crap.

**Comment by IOIIOOIO** [comment]
Thanks @joaokrabbe that worked for me.

This is a very strange problem though because overwriting `$body-font-family` but trying to overwrite `$heading-font-family` has no effect. It confuses me because when I look at the SASS code in Vuetify I see no reason why it wouldn't work as expected.

**Comment by IOIIOOIO** [comment]
I've found a solution on stackoverflow that works.

simply override the `font-family` in the `$headings` map:

**variables.scss**

```
$heading-font-family: 'YourCustomFontFamily';

$headings: (
	'h1': (
		'font-family': $heading-font-family,
	),
	'h2': (
		'font-family': $heading-font-family,
	),
	'h3': (
		'font-family': $heading-font-family,
	),
	'h4': (
		'font-family': $heading-font-family,
	),
	'h5': (
		'font-family': $heading-font-family,
	),
	'h6': (
		'font-family': $heading-font-family,
	),
	'subtitle-1': (
		'font-family': $heading-font-family,
	),
	'subtitle-2': (
		'font-family': $heading-font-family,
	),
	'body-1': (
		'font-family': $heading-font-family,
	),
	'body-2': (
		'font-family': $heading-font-family,
	),
	'caption': (
		'font-family': $heading-font-family,
	),
);
```

**Comment by blackinitial** [comment]
like commented by @loomchild 

read on documentation [vuetify module](https://github.com/nuxt-community/vuetify-module)

> Provide a way to customize Vuetify SASS variables.
> **Only works with [tree-shaking](https://github.com/nuxt-community/vuetify-module#customvariables).**

**Comment by AndrianD** [comment]
> Worked for me!
> 
> **nuxt.config.js**
> 
> ```
> vuetify: {
>   customVariables: ['~/assets/variables.scss'],
>   theme: {
>     dark: false
>   }
> },
> ```
> 
> **variables.scss**
> 
> ```
> $body-font-family: Montserrat;
> 
> .v-application {
>   .display-1,
>   .display-2,
>   .display-3,
>   .display-4,
>   .headline,
>   .title,
>   .subtitle-1,
>   .subtitle-2,
>   .body-1,
>   .body-2,
>   .caption,
>   .overline {
>     font-family: $body-font-family !important;
>   }
> }
> ```

Oh, yes! Thank you. This solution works for me

**Comment by mapkbalaji** [comment]
This worked for me without using NUXT
Thanks @imperatormk for helping out!

**/src/sass/variables.scss**
```
$body-font-family: 'Nunito', Helvetica, sans-serif;

.v-application {
  .display-1,
  .display-2,
  .display-3,
  .display-4,
  .headline,
  .title,
  .subtitle-1,
  .subtitle-2,
  .body-1,
  .body-2,
  .caption,
  .overline {
    font-family: $body-font-family !important;
  }
}
@import '~vuetify/src/styles/settings/_variables.scss';
```

**index.html**
```
<head>
      ....
    <link href="https://fonts.googleapis.com/css?family=Nunito&display=swap" rel="stylesheet">
      ....
</head>
```

**main.js**

`import './sass/variables.scss'`

**Comment by EmadKaizen** [comment]
Playing with [beautify-with-vuetify tutorial](https://github.com/Code-Pop/beautify-with-vuetify/tree/Lesson-9-END)

This problem happens when using npm instead of yarn as package manager

**Comment by jensvlierop** [comment]
#### I found something that worked

I followed the documentation over on the Vuetify website 
https://vuetifyjs.com/en/customization/sass-variables/#example-variable-file

I'm using the CLI plugin and added the following to my `src\scss\variables.scss` file:
```
//all variable changes need to happen before @import
$body-font-family: 'Open Sans', sans-serif !important;

//specificity needs to be taken into consideration for some components
//because !important has been assigned to their css
//https://dev.to/emmabostian/css-specificity-1kca
html,
body,
.v-application {
  font-family: $body-font-family;
}
//added all typography helper classes because of css specificity override
//https://vuetifyjs.com/en/styles/typography#typography 
.v-application {
    .display-4,
    .display-3,
    .display-2,
    .display-1,
    .headline,
    .title,
    .subtitle-1,
    .subtitle-2,
    .body-1,
    .body-2,
    .caption,
    .overline
    {
        font-family: $body-font-family;
    }
}
@import '~vuetify/src/styles/settings/_variables.scss';
@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,400i,600,700,800&display=swap');

**Comment by siddhartharora02** [comment]
So vuetify provides a very simple solution.

In your `src` directory create a `sass`, `scss`, or `styles` directory and then create a file named `variables.scss` or `variables.sass` in it.
> When you run yarn serve or npm run serve, vuetify will automatically hoist the global Vuetify variables to all of your sass/scss files.

Example - `src/scss/variables.scss`

**Below code will change the default body font**
```
@import url('https://fonts.googleapis.com/css2family=Poppins:wght@400;700&display=swap');
$body-font-family: 'Poppins', sans-serif;
```

You can change many more shit in there and can change webpack settings if above doesn't work for you directly - [check this link](https://vuetifyjs.com/en/customization/sass-variables/#sass-variables)

Hope it helps!

**Comment by tleeaf** [comment]
None of these solutions worked for me at first, but then I changed an import statement in vuetify.js and suddenly the font-family changes were successfully applied. 
```
//vuetify.js
//changed from this
//import Vuetify from 'vuetify/lib';

//to this
import Vuetify from 'vuetify/lib/framework';
```
Hope that helps someone!

**Comment by SauceCodeFr** [comment]
> So vuetify provides a very simple solution.
> 
> In your `src` directory create a `sass`, `scss`, or `styles` directory and then create a file named `variables.scss` or `variables.sass` in it.
> 
> > When you run yarn serve or npm run serve, vuetify will automatically hoist the global Vuetify variables to all of your sass/scss files.
> 
> Example - `src/scss/variables.scss`
> 
> **Below code will change the default body font**
> 
> ```
> @import url('https://fonts.googleapis.com/css2family=Poppins:wght@400;700&display=swap');
> $body-font-family: 'Poppins', sans-serif;
> ```
> 
> You can change many more shit in there and can change webpack settings if above doesn't work for you directly - [check this link](https://vuetifyjs.com/en/customization/sass-variables/#sass-variables)
> 
> Hope it helps!

This solution is the more elegant and is working well for me. Just note that you must restart `yarn serve` to take effect !

**Comment by herlambang** [comment]
This works for my case, Vuetify + Nuxt, change global font from Roboto to Open Sans.

Refer to this page: https://github.com/nuxt-community/vuetify-module#defaultassets

nuxt.config.js
```javascript
    treeShake: true,
    defaultAssets: {
      font: {
        family: 'Open Sans' 
      }
    },
```

No need to add custom definition in variables.scss.  
No multiple fonts css embed in page load.

**Comment by Albertbol** [comment]
> This works for my case, Vuetify + Nuxt, change global font from Roboto to Open Sans.
> 
> Refer to this page: https://github.com/nuxt-community/vuetify-module#defaultassets
> 
> nuxt.config.js
> 
> ```js
>     treeShake: true,
>     defaultAssets: {
>       font: {
>         family: 'Open Sans' 
>       }
>     },
> ```
> 
> No need to add custom definition in variables.scss.
> No multiple fonts css embed in page load.

If you want to use only specific font weights or add display swap for example, this solution worked perfectly for me:
```
 defaultAssets: {
      font: {
        family: ['Roboto:400,500,700&display=swap']
      }
    },
```

**Comment by loomchild** [comment]
Thanks for sharing the `defaultAssets` solution, it's great if we have the same font for body and headings! I have updated my short [article](https://medium.com/untitled-factory/changing-default-font-in-vuetify-js-and-nuxt-js-3894e726ff10) to include it.

**Comment by kxhira** [comment]
This works for me with Nuxt.
It doesn't work without `treeShake: true`.

**nuxt.config.js**

```
  vuetify: {
    customVariables: ['~/assets/variables.scss'],
    treeShake: true,
```

**~/assets/variables.scss**

```
$body-font-family: ...;

@import "~vuetify/src/styles/styles.sass";
```

**Comment by mjzarrin** [comment]
I am opening this thread again to give my feedback.
Thank you all Vuetify developers for making amazing library but If I ever knew that simple changing font need too many wasting hours and doing many complex tasks and tricks, I would never picked Vuetify!

what's the meaning of I18n if I couldn't set my language fonts?

My language is RTL and has sticking alphabets which is not support by the default font and the result is ugly and hard to read.

So Please, provide a simple option for changing font!

Thanks

**Comment by mjzarrin** [comment]
For two days I followed all available instructions with no success.
Finally I found the problem is `sass-loader@9.x.x` which is incompatible with Vuetify.
degrade sass-loader to version 8 resolve the problem.

```
yarn remove sass-loader
yarn add sass-loader@8.0.2 --dev
```

**Comment by christianlevesque** [comment]
For anyone using a pure Webpack solution and having trouble with this, you need to have `vuetify-loader` installed in your project *and* set up as a Webpack plugin.

```
// webpack.config.js

const {VuetifyLoaderPlugin} = require("vuetify-loader")

module.exports = {
  // ...
  plugins: [
    // ...
    new VueLoaderPlugin(),
    new VuetifyLoaderPlugin(),
    new CleanWebpackPlugin()
  ]
}
```

I don't know to what extent the order matters, but I put `VuetifyLoaderPlugin` after `VueLoaderPlugin` and immediately before `CleanWebpackPlugin`, and that is working fine for me. But I also got this working about 2 minutes ago, so YMMV.

**Comment by farajist** [comment]
I managed to apply a custom font to headings by using the `.text-{breakpoint}-{value}` classes instead, where `value` is in (h1, h2 ... etc.) and `breakpoint` can be omitted.

See docs : [Typography](https://vuetifyjs.com/en/styles/text-and-typography/#typography)

**Comment by Siddharth-Ashri** [comment]
> > I had to declare my custom font-family **before** importing the default vuetify style in my custom main.scss. That way vuetify properly picks up the custom definition:
> > ```
> > $body-font-family: ...;
> > 
> > @import "~vuetify/src/styles/styles.sass";
> > ```
> 
> In case anyone else is reading this, this definitely worked for me.

I have been trying to import .sass extension into the .scss file webpack crashes when trying to compile the files.

**Comment by YasuroNN** [comment]
Global overriding is easy, but how can i override font-family local for each component?

**Comment by Yeganloo** [comment]
Same problem and non of the solution works for me!
I found something odd!
If i pass the vuetify config inside nuxt.config.js, all the above solution working but if i use optionsPath, (and i have to use it) non of the solutions work!
Any idea why this happens?

**Comment by Yeganloo** [comment]
Just found it:
https://github.com/nuxt-community/vuetify-module/issues/245

**Comment by jonaslocke** [comment]
This worked for me:

Just followed the instructions on documentation guide and created a file on src/sass/variables.scss.
[https://vuetifyjs.com/en/features/sass-variables/#vue-cli-install](url)

inside that file, used this:

`@import "~vuetify/src/styles/settings/_variables";

$body-font-family: "Catamaran";
$headings-font-family: "Montserrat";

$typoOptions: "display-4", "display-3", "display-2", "display-1", "headline",
  "title", "subtitle-1", "subtitle-2", "body-1", "body-2", "caption", "overline";

$headings-breakpoint: "sm", "md", "lg", "xl";
$headings-value: "h1", "h2", "h3", "h4", "h5", "h6", "subtitle-1", "subtitle-2",
  "body-1", "body-2", "button", "caption", "overline";
#app {
  font-family: $body-font-family, sans-serif !important;
  @each $typoOption in $typoOptions {
    .#{$typoOption} {
      font-family: $headings-font-family, sans-serif !important;
    }
  }
  @each $heading in $headings-value {
    @each $breakpoint in $headings-breakpoint {
      .text-#{$breakpoint}-#{$heading} {
        font-family: $headings-font-family, sans-serif !important;
      }
    }
  }
}
`

This way you override all the classes generated by vuetify.

**Comment by jsun969** [comment]
I found a very easy solution
Just add **!important** after `font-family`
My style is
```scss
@font-face {
  font-family: 'Sarasa Mono SC';
  src: url('./assets/sarasa-mono-sc-regular.ttf') format('truetype');
}
#message {
  font-family: 'Sarasa Mono SC' !important;
}
```

**Comment by simeon9696** [comment]
> I've found a solution on stackoverflow that works.
> 
> simply override the `font-family` in the `$headings` map:
> 
> **variables.scss**
> 
> ```
> $heading-font-family: 'YourCustomFontFamily';
> 
> $headings: (
> 	'h1': (
> 		'font-family': $heading-font-family,
> 	),
> 	'h2': (
> 		'font-family': $heading-font-family,
> 	),
> 	'h3': (
> 		'font-family': $heading-font-family,
> 	),
> 	'h4': (
> 		'font-family': $heading-font-family,
> 	),
> 	'h5': (
> 		'font-family': $heading-font-family,
> 	),
> 	'h6': (
> 		'font-family': $heading-font-family,
> 	),
> );
> ```

Sadly this didn't work for me. The headings still take my body font: 
```
@import '@fontsource/ledger/latin.css';
@import '@fontsource/raleway/latin.css';

$font-size-root: 16px;
$body-font-family: Raleway, Helvetica, sans-serif;
$heading-font-family: serif;
$headings: (
	'h1': (
		'font-family': $heading-font-family,
	),
	'h2': (
		'font-family': $heading-font-family,
	),
	'h3': (
		'font-family': $heading-font-family,
	),
	'h4': (
		'font-family': $heading-font-family,
	),
	'h5': (
		'font-family': $heading-font-family,
	),
	'h6': (
		'font-family': $heading-font-family,
	),
);
@import '~vuetify/src/styles/styles.sass';
```

**Comment by gautam-patadiya** [comment]
@joaokrabbe thanks! you are genius

**Comment by KaelWD** [comment]
@simeon9696 It only applies to the `text-h*` classes, not `h*` elements.

**Comment by ghRocam** [comment]
Ok, after doing a lot of tweaks, I concluded this should be change to something more intuitive or at least more descriptive, case in point:
//my file is src/sass/variables.scss

$body-font-family: 'IBM Plex Sans', sans-serif;
$heading-font-family: 'Exo', sans-serif;

$headings: (
  (
    // this object properties are being applied to the display-4 class
    'h1': (
      'size': 6rem,
      'weight': 300,
      'line-height': 6rem,
      'letter-spacing': -.015625em,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // display-3 class
    'h2': (
      'size': 3.75rem,
      'weight': 300,
      'line-height': 3.75rem,
      'letter-spacing': -.0083333333em,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // display-2 class
    'h3': (
      'size': 3rem,
      'weight': 400,
      'line-height': 3.125rem,
      'letter-spacing': normal,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // display-1 class
    'h4': (
      'size': 2.125rem,
      'weight': 400,
      'line-height': 2.5rem,
      'letter-spacing': .0073529412em,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // headline class
    'h5': (
      'size': 1.5rem,
      'weight': 400,
      'line-height': 2rem,
      'letter-spacing': normal,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // title class
    'h6': (
      'size': 1.25rem,
      'weight': 500,
      'line-height': 2rem,
      'letter-spacing': .0125em,
      'font-family': $heading-font-family,
      'text-transform': false
    ),
    // subtitle-1 class
    'subtitle-1': (
      'size': 1rem,
      'weight': normal,
      'line-height': 1.75rem,
      'letter-spacing': .009375em,
      'font-family': $body-font-family,
      'text-transform': false
    ),
    // subtitle-2 class
    'subtitle-2': (
      'size': .875rem,
      'weight': 500,
      'line-height': 1.375rem,
      'letter-spacing': .0071428571em,
      'font-family': $body-font-family,
      'text-transform': false
    ),
    // body-1 class
    'body-1': (
      'size': 1rem,
      'weight': 400,
      'line-height': 1.5rem,
      'letter-spacing': .03125em,
      'font-family': $body-font-family,
      'text-transform': false
    ),
    // body-2 class
    'body-2': (
      'size': .875rem,
      'weight': 400,
      'line-height': 1.25rem,
      'letter-spacing': .0178571429em,
      'font-family': $body-font-family,
      'text-transform': false
    ),
    // text-button class
    'button': (
      'size': .875rem,
      'weight': 500,
      'line-height': 2.25rem,
      'letter-spacing': .0892857143em,
      'font-family': $body-font-family,
      'text-transform': uppercase
    ),
    // caption class
    'caption': (
      'size': .75rem,
      'weight': 400,
      'line-height': 1.25rem,
      'letter-spacing': .0333333333em,
      'font-family': $body-font-family,
      'text-transform': false
    ),
    // overline class
    'overline': (
      'size': .75rem,
      'weight': 500,
      'line-height': 2rem,
      'letter-spacing': .1666666667em,
      'font-family': $body-font-family,
      'text-transform': uppercase
    )
  )
);

other solutions pointed above are not ideal because duplicate massively styles all around, with above at least you can have one style only but the thing is just awful to track which object is being applied to which class, just awfuuul i repeat

so, in summary: 

//    class -> map object name
//   .title -> h6
//   .headline -> h5
//   .display-1 -> h4
//   .display-2 -> h3
//   .display-3 -> h2
//   .display-4 -> h1
//   .subtitle-1 -> subtitle-1 yay
//   .subtitle-2 -> subtitle-2
//   .body-1 -> body-1
//   .body-2 -> body-2
//   .caption -> caption
//   .overline -> overline
//   .text-button -> button

I hope this helps someone else

**Comment by DiegoPerea** [comment]
Even if you customize your variables like the examples above, the sass adds the !important at the end, preventing more than 1 custom font because the !important in vuetify overrides !important of user css

Let's say your project needs 2 custom fonts. You'd like to write your classes like: .h1.font1 and .h1.font2 to use the same styles of vuetify's h1 and override the default font with your custom font1 or font2 when needed. But vuetify adds !important so it will choose the same font when using .h1 

In other words if you customize your variables like the example above:

> ```
> $body-font-family: sans-serif;
> $heading-font-family: sans-serif;
> 
> $headings: (
>   (
>     // this object properties are being applied to the display-4 class
>     'h1': (
>       'size': 6rem,
>       'weight': 300,
>       'line-height': 6rem,
>       'letter-spacing': -.015625em,
>       'font-family': $heading-font-family,
>       'text-transform': false
>     ),
> ```

Ends up bringing

```
.h1 {
  ...
  font-family: sans-serif !important
  letter-spacing: -.015625em !important
}
```

So you have to write a custom h1 for every other font you have in the project.

The real solution would be that the css generated by vuetify sass variables shouldn't have the !important flag in any typo styles like font-family and letter spacing etc... so they can be overrided easily with custom css and !important

**Comment by jelicich** [comment]
I also struggled to change the font family, some of the answers here didn't work and some others were not the solution I wanted to implement.

**I managed to change the font** using sass variables by using [vuetify-loader](https://github.com/vuetifyjs/vuetify-loader) to build the project using the uncompiled styles from vuetify. If you are loading the final styles (`vuetify/dist/vuetify.min.css`) you won't be able to overwrite any sass variable because they won't exist at that point.

I'm using yarn so: `yarn add vuetify-loader --dev`.

Since I'm using `vuetify-loader` now, I no longer need the minified css, so I removed it and changed my Vuetify import to `vuetify/lib` as it states in the [docs](https://vuetifyjs.com/en/features/sass-variables/):

`import Vuetify from 'vuetify/lib';`

Once I had that I also needed to add the following properties to my `vue.config.js` to make Vuetify components available globally:

```
const VuetifyLoaderPlugin = require('vuetify-loader/lib/plugin')

...
configureWebpack: {
  plugins: [
    new VuetifyLoaderPlugin(),
  ],
},
...
```

Finally, I had to solve some compiling issues, I was using `node-sass` and I read somewhere that `sass` should be used instead (I don't recall where I read it, sorry). Also it is important the version: `"sass": "~1.32.12",`. Although the newer versions will work, they will throw warnings regarding the way vuetify does some math calculations (It uses `/` and it is deprecated). So in order to avoid those warnings the version mentioned before is recommended.

With this config I was able to set my variables in `src/styles/variables.scss` and overwrite Vuetify, in my opinion this the right way to do it as it won't require to target elements with specific selectors or use the `!important` statement. 

An example of my `variables.scss`

```
// Overwrite Vuetify defaults
// Globals
$body-font-family: 'Inter', sans-serif;
$heading-font-family: 'Inter', sans-serif;
$border-radius-root:	8px;
$shadow-key-ambient-opacity: rgba(51, 51, 51, 0.25);
$shadow-key-penumbra-opacity: rgba(51, 51, 51, 0.25);
$shadow-key-umbra-opacity: rgba(51, 51, 51, 0.25);

// v-btn
$btn-border-radius: 4px;
```

PS: It is important to mention that using vuetify-loader and Vuetify uncompiled styles, the build time increases a bit. Every time you make a change, it takes a bit longer to compile and show your changes.

**Comment by JeremyQtweb** [comment]
Let me know when someone has got it working in Nuxt 3 / Vuetify 3 cos right now absolutely nothing worked extrapolating from this thread or the official docs (although half of that doesn't make sense because it's all snippets or doesn't seem to be shown in a Nuxt compatible fashion). It's been an afternoon of futility with random build errors that seem to have no bearing or the font just doesn't change. This is still madness, I thought this latest generation of frameworks was supposed to make things easier.

**Comment by andoshin11** [comment]
This is crazy. Why do you even need `!important` for a UI library? 🤔

**Comment by anaelbe** [comment]
Using a vite + typescript stack, managed to make it work in Vuetify 3 inside `settings.scss` with:

```scss
@use "vuetify" with (
    $body-font-family: CustomFontName,
    $heading-font-family: CustomFontName
);
```

In `vite.config.ts`, you should have a link to settings :

```ts
export default defineConfig({
  plugins: [
    vuetify({
      styles: {
        configFile: 'src/styles/settings.scss',
      },
    }),
  ],
```

**Comment by xaviqo** [comment]
I like Vuetify, but seriously, it can't be that changing the font is such a pain and it's not well documented on their site.

**Comment by lerryklitzke** [comment]
For Vuetify 3 + Nuxt 3
assets/styles/main.scss
`
$custom-font-family: monospace;
@use 'vuetify' with (
  $body-font-family: $custom-font-family
);
`
plugins/vuetify.ts
`
import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';
import '@/assets/styles/main.scss';

export default defineNuxtPlugin((nuxtApp) => {
  const vuetify = createVuetify({
    ssr: true,
    components,
    directives,
  });
  nuxtApp.vueApp.use(vuetify);
});
`

Bonus: if you want to use a font from Google Fonts, you can do the following:
- Create this file: assets/styles/fonts.scss with your fonts
`
`
-

**Comment by bakman2** [comment]
Roboto is a horrible font - and reading the pain above of changing it means I will never use this project. Same applies to the massive paddings in form fields, they are extremely inefficient - while there no simple way to get rid of them, is this 2024 ?

**Comment by MartinX3** [comment]
I moved to quasar and I'm happier now.

**Comment by s3rgeym** [comment]
Instruction for vuetify 3 and vite

Install Vuetify plugin for Vite:

```bash
npm install -D vite-plugin-vuetify
```

Configure Vuetify to use the custom font family:

`src/styles/settings.scss`
```scss
$custom-font-family: 'Open Sans', monospace;

@use 'vuetify/settings' with (
  $body-font-family: $custom-font-family,
  $heading-font-family: $custom-font-family
);
```

Configure Vite to use the custom font settings:

`vite.config.js`
```js
import vuetify from 'vite-plugin-vuetify'

export default defineConfig({
  plugins: [
    vuetify({
      styles: {
        configFile: 'src/styles/settings.scss'
      }
    })
    // ...
  ]
  // ...
})
```

Other files:

`src/styles/fonts.scss`
```scss
@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@300;400;700&display=swap');
```

`src/styles/main.scss`
```scss
@use './fonts';
// ...
```

`main.js`
```js
// ...
import './styles/main.scss'
// ...
```
